src/components/pages/profile/account/SetMFA.vue

```diff
@@ -0,0 +1,371 @@
+  isEnabled.value = false;
+  resetState();
+};
+
+/** Called when the slider changes */
+const onSlideChange = (details: AppSliderSlideEventDetails) => {
+  syncStepperWithSlider(details);
+  validateStep(details);
+};
+
+/** Reset / init the state of the component */
+const resetState = () => {
+  appSliderCo.value?.toSlide(0);
+  appStepper.value.selectedStepIndex = 0;
+  appStepper.value.steps.forEach((step) => {
+    step.validated = false;
+    step.error = false;
+  });
+  userToken.value = '';
+  userTokenValid.value = false;
+};
+
+/** Sync the stepper with the slider */
+const syncStepperWithSlider = (details: AppSliderSlideEventDetails) => {
+  if (appStepper.value.selectedStepIndex === details.slideNumber) {
+    return;
+  }
+  appStepper.value.selectedStepIndex = details.slideNumber;
+
+  validatePreviousSteps();
+};
+
+/** Validate the current step */
+const validateStep = (details: AppSliderSlideEventDetails) => {
+  if (details.slideNumber === 2) {
+    validateTOTOToken();
+  }
+};
+
+/** Validate all previous steps */
+const validatePreviousSteps = () => {
+  for (let i = 0; i < appStepper.value.steps.length; i++) {
+    const step = appStepper.value.steps[i];
+    step.validated = i < appStepper.value.selectedStepIndex;
+  }
+};
+
+/** Called when the stepper changes */
+const onStepperChange = (stepIndex: number) => {
+  appSliderCo.value?.toSlide(stepIndex);
+  validatePreviousSteps();
+};
+
+/** Called when the totp identity changes */
+const onTotpIdentityChanged = (newValue: TotpIdentity) => {
+  totpSecret.value = new Secret();
+  totp = new TOTP({
+    issuer: newValue.issuer,
+    label: newValue.label,
+    secret: totpSecret.value,
+  });
+  totpUrl.value = totp.toString();
+};
+
+/** Validate the totp token / user input code */
+const validateTOTOToken = (window = 2) => {
+  if (!totp) {
+    throw new Error('totp is null!');
+  }
+  const validateStep = appStepper.value.steps[1];
+  const resultStep = appStepper.value.steps[2];
+  const delta = totp.validate({
+    token: userToken.value,
+    window,
+  });
+
+  userTokenValid.value = delta === 0;
+
+  validateStep.error = !userTokenValid.value;
+  resultStep.error = !userTokenValid.value;
+  validateStep.validated = userTokenValid.value;
+  resultStep.validated = userTokenValid.value;
+
+  return userTokenValid.value;
+};
+
+/** Are all steps done with no errors? */
+const allStepsDone = computed(() => {
+  return (
+    appStepper.value.steps[0].validated &&
+    !appStepper.value.steps[0].error &&
+    appStepper.value.steps[1].validated &&
+    !appStepper.value.steps[1].error &&
+    appStepper.value.steps[2].validated &&
+    !appStepper.value.steps[2].error
+  );
+});
+
+/** Fetch the contact and set the totp identity */
+watch(
+  toRef(props, 'contactId'),
+  async (contactId) => {
+    const contact = await fetchContact(contactId, ['profile']);
+    totpIdentity.value.issuer = 'beabee'; // TODO: Use name of beabee instance
```

Comment:
You should be able to use `generalContent.value.organisationName` for this (or something like that)

Reply:
thanks